app/services/fact_cleaner.rb

```diff
@@ -39,11 +43,42 @@ def find_compose_orphaned_facts(found)
   end
 
   def delete!(to_delete)
-    logger.info "#{to_delete.size} facts will be removed"
+    logger.debug "#{to_delete.size} facts will be removed"
     to_delete.each do |fact|
       logger.debug { "Removing fact #{fact}" }
       fact.destroy
     end
-    logger.info "Cleanup of facts finished!"
+    logger.debug "Cleanup of facts finished!"
+  end
+
+  def find_excluded_facts
+    logger.debug "Searching for facts that match excluded pattern..."
+    fact_name_ids = []
+
+    excluded_facts.each do |exclude_pattern|
+      sql_pattern = exclude_pattern.gsub('*', '%')
+      # for facts that look like ignore_me::something
+      fact_name_ids += FactName.unscoped.where('name LIKE ?', sql_pattern).pluck(:id)
+      fact_name_ids += FactName.unscoped.where('name LIKE ?', sql_pattern + '::%').pluck(:id)
+      # for facts that look like common::ignore_me::something
+      fact_name_ids += FactName.unscoped.where('name LIKE ?', '%::' + sql_pattern).pluck(:id)
+      fact_name_ids += FactName.unscoped.where('name LIKE ?', '%::' + sql_pattern + '::%').pluck(:id)
+    end
+    fact_name_ids
+  end
```

Comment:
Is this method new in the last commit? I haven't seen it. Anyway, this will generate *huge* SQL query and what's the worst - the server will need to run it for each individual fact upload. The cost for this consistency is way too high, I suggest to move this into rake task that can be run from cron or something once a week. This has potential to slow down fact import, we also haven't tested performance of this on MySQL.

Reply:
I realized this method is only called from clean! method, so that is fine.